experimental/priority_sampler.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package experimental
+
+import (
+	"sync/atomic"
+
+	"github.com/uber/jaeger-client-go"
+)
+
+// prioritySamplerState keeps the state of all underlying samplers, specifically
+// whether each of them has previously returned retryable=false, in which case
+// those samplers are no longer invoked on future sampling calls.
+type prioritySamplerState struct {
+	samplerFiredBitmap []uint64
+}
+
+func newPrioritySamplerState(count int) *prioritySamplerState {
```

Comment:
I thought about it, but is it really going to simplify that much?  I'd need to validate & handle errors, vs. having `/64` and `% 64` in a couple places. One pro of 64 limit is it would save one malloc.